src/humancellatlas/data/metadata/api.py

```diff
@@ -600,6 +601,11 @@ class ManifestEntry:
     version: str
 
     def __init__(self, json: JSON):
+        # '/' was once forbidden in file paths and was encoded with '!'. Now
+        # '/' is allowed and we force it in the metadata so that backwards
+        # compatibility is simplified downstream.
+        json = cast(dict, json)
```

Comment:
Why cast here but not for File?

Reply:
For future reference, cast() does nothing at runtime. An assert would have been better. But:

We should change the definition of `JSON` to correctly reflect the mutability,

Reply:
Cast removed; JSON modified.